From ea62445940924aff861314da242c1c32aeb3fe1c Mon Sep 17 00:00:00 2001
From: De Huo <De.Huo@windriver.com>
Date: Thu, 8 Aug 2019 18:09:40 +0800
Subject: [PATCH] Rule cracklib_accounts_password_pam_dcredit add oval check

For rule cracklib_accounts_password_pam_dcredit add oval check

Upstream-Status: Pending

Signed-off-by: De Huo <De.Huo@windriver.com>
---
 .../bash/wrlinux1019.sh                            | 14 ++++++++++
 .../oval/wrlinux1019.xml                           | 32 ++++++++++++++++++++++
 2 files changed, 46 insertions(+)
 create mode 100644 linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_dcredit/bash/wrlinux1019.sh
 create mode 100644 linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_dcredit/oval/wrlinux1019.xml

diff --git a/linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_dcredit/bash/wrlinux1019.sh b/linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_dcredit/bash/wrlinux1019.sh
new file mode 100644
index 0000000..d456e0a
--- /dev/null
+++ b/linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_dcredit/bash/wrlinux1019.sh
@@ -0,0 +1,14 @@
+# platform = multi_platform_wrlinux
+
+. /usr/share/scap-security-guide/remediation_functions
+populate var_password_pam_dcredit
+
+if grep -q '^[^#].*pam_cracklib.so' /etc/pam.d/common-password; then
+  if grep -q 'dcredit=' /etc/pam.d/common-password; then
+	  sed -i "s/dcredit=[0-9]*/dcredit=$var_password_pam_dcredit/" /etc/pam.d/common-password
+  else
+	  sed -i "/password.*pam_cracklib.so/s/$/ dcredit=$var_password_pam_dcredit/" /etc/pam.d/common-password
+  fi
+else
+  echo "password  required pam_cracklib.so dcredit=$var_password_pam_dcredit" >>/etc/pam.d/common-password
+fi
diff --git a/linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_dcredit/oval/wrlinux1019.xml b/linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_dcredit/oval/wrlinux1019.xml
new file mode 100644
index 0000000..bd4baec
--- /dev/null
+++ b/linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pamcracklib/cracklib_accounts_password_pam_dcredit/oval/wrlinux1019.xml
@@ -0,0 +1,32 @@
+<def-group>
+  <definition class="compliance" id="cracklib_accounts_password_pam_dcredit" version="1">
+    <metadata>
+      <title>Set Password dcredit Requirements</title>
+      <affected family="unix">
+        <platform>multi_platform_wrlinux</platform>
+      </affected>
+      <description>The password dcredit should meet minimum requirements using pam_cracklib</description>
+    </metadata>
+    <criteria>
+      <criterion comment="Conditions for dcredit are satisfied" test_ref="test_password_pam_cracklib_dcredit" />
+    </criteria>
+  </definition>
+
+  <ind:textfilecontent54_test check="all" comment="check the configuration of /etc/pam.d/common-password" id="test_password_pam_cracklib_dcredit" version="1">
+    <ind:object object_ref="obj_password_pam_cracklib_dcredit" />
+    <ind:state state_ref="state_password_pam_cracklib_dcredit" />
+  </ind:textfilecontent54_test>
+  <ind:textfilecontent54_object id="obj_password_pam_cracklib_dcredit" version="1">
+    <ind:path>/etc/pam.d</ind:path>
+    <ind:filename>common-password</ind:filename>
+    <ind:pattern operation="pattern match">^password[\s]+(?:requisite|required)[\s]+(?:/lib/security/(?:\$ISA/)*)*pam_cracklib.so[\w_\.\-=\s]*[\s]+dcredit=(-?\d+)(?:[\s]|$)</ind:pattern>
+    <ind:instance datatype="int" operation="less than or equal">1</ind:instance>
+  </ind:textfilecontent54_object>
+  <ind:textfilecontent54_state id="state_password_pam_cracklib_dcredit" version="1">
+    <ind:instance datatype="int">1</ind:instance>
+    <ind:subexpression datatype="int" operation="less than or equal" var_ref="var_password_pam_dcredit" />
+  </ind:textfilecontent54_state>
+
+  <external_variable comment="External variable for pam_cracklib dcredit" datatype="int" id="var_password_pam_dcredit" version="1" />
+
+</def-group>
-- 
1.9.1

